### PR TITLE
Add functions for application wise log levels

### PR DESCRIPTION
**Pull request by hauleth**
Ref #8414

### CODE REVIEW

**Comment by hauleth** [comment]
I am preparing PR with fix for the test issues right now.

File: lib/logger/lib/logger.ex [hunk 1]
```diff
@@ -661,6 +661,32 @@ defmodule Logger do
...
+  Puts logging level for modules in given application.
+
+  This will take priority over the primary level set, so it can be
+  used to increase or decrease verbosity of some parts of the project.
+
+  Equivalent of:
+
+      appname |> Application.spec(:modules) |> Logger.put_module_level(level)
+  """
+  @doc since: "1.12.0"
```
Comment (josevalim) [inline lib/logger/lib/logger.ex:674]:
```suggestion
  @doc since: "1.13.0"
```

File: lib/logger/lib/logger.ex [hunk 2]
```diff
@@ -661,6 +661,32 @@ defmodule Logger do
...
+  defdelegate put_application_level(appname, level), to: :logger, as: :set_application_level
+
+  @doc """
+  Deletes logging level for all modules in given application to primary level
+
+  Equivalent of:
+
+      appname |> Application.spec(:modules) |> Logger.delete_module_level()
+  """
+  @doc since: "1.12.0"
```
Comment (josevalim) [inline lib/logger/lib/logger.ex:685]:
```suggestion
  @doc since: "1.13.0"
```

**Comment by josevalim** [comment]
@hauleth the failures you see here seem to be heisenbugs, since they don't fail on master. git is also emit warnings but that does not have an impact on the test suite.

In any case, this looks great, thanks! We just need tests and we can ship it!

**Comment by hauleth** [comment]
@josevalim I do not really have an idea how we could test `Logger.put_application_level/2`, as this would require to have example `app.src` that could be used. Any idea how I can do such stunt?

**Comment by josevalim** [comment]
@hauleth you can use the ExUnit app as an examples (or the EEx one).

**Comment by josevalim** [comment]
To be clear, for the put_application_level, you can simply write a unit test that checks it will set `put_module_level` correctly, as you have tested the put_module_level version. :)

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**